vms/platformvm/txs/executor/camino_tx_executor.go

```diff
@@ -353,22 +354,90 @@ func (e *CaminoStandardTxExecutor) ExportTx(tx *txs.ExportTx) error {
 		return err
 	}
 
-	for _, output := range tx.ExportedOutputs {
+	if err := e.StandardTxExecutor.ExportTx(tx); err != nil {
+		return err
+	}
+
+	if err := e.wrapAtomicElementsForMultisig(tx); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (e *CaminoStandardTxExecutor) wrapAtomicElementsForMultisig(tx *txs.ExportTx) error {
+	putReqs := e.AtomicRequests[tx.DestinationChain].PutRequests
+	for i, output := range tx.ExportedOutputs {
 		out, ok := output.Out.(*secp256k1fx.TransferOutput)
 		if !ok {
 			return locked.ErrWrongOutType
 		}
-		for _, addr := range out.Addrs {
-			_, err := e.State.GetMultisigAlias(addr)
-			if err != nil && err != database.ErrNotFound {
-				return err
-			} else if err == nil {
-				return errMsigAlias
-			}
+		if i >= len(putReqs) {
+			return errors.New("msig position out of range of put requests")
+		}
+
+		aliasInfs, err := e.Fx.CollectMultisigAliases(&out.OutputOwners, e.State)
+		if err != nil {
+			return err
+		}
+		if len(aliasInfs) == 0 {
+			// no need to wrap current UTXO
+			continue
+		}
+
+		req := putReqs[i]
+		var utxo avax.UTXO
+		_, err = txs.Codec.Unmarshal(req.Value, &utxo)
+		if err != nil {
+			return err
 		}
+
+		// wrap utxo with alias
+		aliases, err := sortAliases(aliasInfs)
+		if err != nil {
+			return err
+		}
+
+		wrappedUtxo := &avax.UTXOWithMSig{
+			UTXO:    utxo,
+			Aliases: aliases,
+		}
+		bytes, err := txs.Codec.Marshal(txs.Version, wrappedUtxo)
+		if err != nil {
+			return err
+		}
+
+		putReqs[i] = &atomic.Element{
+			Key:    req.Key,
+			Value:  bytes,
+			Traits: req.Traits,
+		}
+	}
+
+	return nil
+}
+
+func sortAliases(alinf []interface{}) ([]verify.State, error) {
+	if len(alinf) == 0 {
+		return []verify.State{}, nil
+	}
+
+	aliases := make([]*multisig.AliasWithNonce, len(alinf))
+	for i, inf := range alinf {
+		ali, ok := inf.(*multisig.AliasWithNonce)
+		if !ok {
+			return nil, errors.New("wrong type, expected multisig.AliasWithNonce")
```

Comment:
...maybe this still makes sens - as I'm relying on `Fx.CollectMultisigAliases(...)` return.